src/main/app/forms/models/multiRowForm.ts

```diff
@@ -0,0 +1,43 @@
+import { Serializable } from 'models/serializable'
+import { ItemInMultiRowForm } from 'forms/models/itemInMultiRowForm'
+
+export abstract class MultiRowForm<T extends ItemInMultiRowForm> implements Serializable<MultiRowForm<T>> {
+
+  rows: T[]
+
+  abstract getMaxNumberOfRows (): number
+
+  abstract createEmptyRow (): T
+
+  abstract deserializeRows (rows: any): T[]
+
+  deserialize (input?: any): MultiRowForm<T> {
```

Comment:
Why does it need to be an instance method instead of a static one?

Reply:
This is just the design we already have in 100% of our models. 

I my humble opinion this could (should) be static. BUT as we are where we are I'd not change it now. 0 (zero) business value a lot of work.

Reply:
@geek-soft